internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -354,6 +356,40 @@ func (s *Server) registerWebhookForRepository(
 	return regResult, nil
 }
 
+func (s *Server) deleteWebhookFromRepository(
+	ctx context.Context,
+	provider db.Provider,
+	projectID uuid.UUID,
+	dbrepo db.Repository,
+) error {
+	pbOpts := []providers.ProviderBuilderOption{
+		providers.WithProviderMetrics(s.provMt),
+	}
+	providerBuilder, err := providers.GetProviderBuilder(ctx, provider, projectID, s.store, s.cryptoEngine, pbOpts...)
+	if err != nil {
+		return status.Errorf(codes.Internal, "cannot get provider builder: %v", err)
+	}
+
+	client, err := providerBuilder.GetGitHub(ctx)
+	if err != nil {
+		return status.Errorf(codes.Internal, "cannot create github client: %v", err)
+	}
+
+	webhookId := dbrepo.WebhookID
+	if webhookId.Valid {
+		resp, err := client.DeleteHook(ctx, dbrepo.RepoOwner, dbrepo.RepoName, int64(webhookId.Int32))
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusNotFound {
+				// if the hook is not found, we can ignore the error, user might have deleted it manually
+				return nil
+			}
+			return status.Errorf(codes.Internal, "cannot delete webhook: %v", err)
```

Comment:
I was going to ask if you use `%w` here and above to wrap the error, but looking at https://github.com/grpc/grpc-go/issues/3616, it looks like the GRPC team rejected supporting `Unwrap` as it might lead to expectations that it would work between client and server.